code/game/objects/items/RPD.dm

```diff
@@ -471,27 +472,46 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 				return
 			// Check for differences in unconnected directions
 			var/target_differences = (p_init_dir ^ target_init_dir) & ~S.connections
-			if (target_differences)
-				to_chat(user, span_notice("You start reprogramming \the [S]..."))
-				playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE)
-				if(do_after(user, reprogram_speed, target = S))
-					// Double check to make sure that nothing has changed. If anything we were about to change is now connected, abort
-					if (target_differences & S.connections)
-						to_chat(user, span_warning("\The [src]'s screen flashes a warning: Can't configure a pipe in a currently connected direction."))
-						return
-					var/new_dir = (S.GetInitDirections() & ~target_differences) | target_differences
-					// Don't make a smart pipe with only one connection
-					if (ISSTUB(new_dir))
-						to_chat(user, span_warning("\The [src]'s screen flashes a warning: Can't configure a pipe to only connect in one direction."))
-						return
-					S.SetInitDirections(new_dir)
-					S.update_pipe_icon()
-					user.visible_message(span_notice("[user] reprograms the \the [S]."),span_notice("You reprogram \the [S]."))
+			if (!target_differences)
+				to_chat(user, span_warning("\The [S] is already in this configuration for its unconnected directions!"))
+				return
+			to_chat(user, span_notice("You start reprogramming \the [S]..."))
+			playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE)
+			if(!do_after(user, reprogram_speed, target = S))
+				return
+			// Double check to make sure that nothing has changed. If anything we were about to change is now connected, abort
+			if (target_differences & S.connections)
+				to_chat(user, span_warning("\The [src]'s screen flashes a warning: Can't configure a pipe in a currently connected direction."))
+				return
+			var/old_dir = S.GetInitDirections()
```

Comment:
Why not rename/reuse target_init_dir for this?
Oh and by the by, if you want to merge this and your other smart pipe fix pr, feel free. Respect the gbp grind tho, so not a requirement?

Reply:
That's a good idea. The intent was to split up the PRs, so we don't end up an a situation similar to having fixes for "AAAA I just want to replace the Metastation Incinerator gas pumps with manual valves, without having to rip up the reinforced floor and replace the smart pipes underneath AAAA" gated behind reviewing sweeping changes and additions to functionality for a week. That's reasonable for not being able to see vent/scrubber IDs, but less so for similar changes to the same section of the same file.